src/App/VPSScan/Scan/RunS3Upload.hs

```diff
@@ -0,0 +1,109 @@
+module App.VPSScan.Scan.RunS3Upload
+  (execS3Upload
+  )
+where
+
+import qualified Aws
+import qualified Aws.S3 as S3
+import qualified Aws.Core as AwsCore
+import Control.Monad.Trans.Resource
+import Control.Effect.Lift
+import Network.HTTP.Conduit (newManager, Manager, tlsManagerSettings, RequestBody(..))
+import Path.IO (listDirRecur, isSymlink)
+import Control.Monad.Except
+import qualified Data.ByteString as S
+import Control.Carrier.TaskPool
+import Effect.Logger
+import GHC.Conc.Sync (getNumCapabilities)
+import Text.URI
+import qualified Data.Text.Encoding as TE
+
+import System.IO
+import Control.Applicative
+import qualified Data.Text as T
+import Control.Carrier.Trace.Printing
+import App.VPSScan.Types
+import Control.Effect.Diagnostics
+import Prologue
+
+data S3UploadError = ErrorRunningS3Upload Text
+  deriving (Eq, Ord, Show, Generic, Typeable)
+
+instance ToDiagnostic S3UploadError where
+  renderDiagnostic = \case
+    ErrorRunningS3Upload err -> "Error while uploading to S3: " <> pretty err
+
+-- S3.S3Configuration is defined here:
+-- https://github.com/aristidb/aws/blob/9ccfce8c50723e1b7e02ba59a9d38227e938ac0a/Aws/S3/Core.hs#L69
+s3config :: IPROpts -> S3.S3Configuration Aws.NormalQuery
+s3config IPROpts{..} =
```

Comment:
[nit] it's hard to tell what fields you're using here from IPROpts -- if it's just the `URI`, can we make this call out to a new top-level function that has a simpler type? `s3configFromURI :: URI -> S3.S3Configuration Aws.NormalQuery` or similar